src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -296,6 +296,22 @@ private void getGitVersion() {
         return gitVersion >= requestedVersion;
     }
 
+    /**
+     * Compare the current cli git version with the required version.
+     * Finds if the current cli git version is at-least the required version
+     *
+     * Returns True if the current cli git version is al-least the required version
+     *
+     * @param major ,required major version for cli git
+     * @param minor ,required minor version for cli git
+     * @param rev ,required revision for cli git
+     * @param bugfix ,required patches for cli git
+     * @return true if the cli git version is al-least the required version
+     **/
+    public boolean compareLeastGitVersion(int major, int minor, int rev, int bugfix) {
```

Comment:
I believe that Java compare functions are generally expected to return an int with values less than zero meaning the current object is "less" than the argument, zero meaning that the current object is the same as the argument, and greater than zero if the current object is "greater" than the argument.  See the [String.compareTo javadoc](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#compareTo-java.lang.String-] for more details.

I think that you want a method name that uses the word `is` rather than `compare` or you need to create a new implementation that does a comparison.  I believe that I used `is` in the original definition because I didn't envision a time when we would need to sort a list of git implementations by their version number.  I don't think we need to sort git implementations by version number for this case, though I could be misunderstanding your need for this API.

If we need ordering / sorting of CliGitAPIImpl objects, then it is probably best to implement the [Comparable interface](https://docs.oracle.com/javase/8/docs/api/java/lang/Comparable.html)
```suggestion
    public boolean compareLeastGitVersion(int major, int minor, int rev, int bugfix) {
```

I'd also be fine if you changes isAtLeastVersion from a private method to a public method.  If the `isAtLeastVersion` name is not clear enough, then adding a new public method with a better name is also fine.